<?xml version="1.0"?>
<launch>
  <arg name="uri1" default="radio://0/46/2M/E7E7E7E701" />
  <arg name="uri2" default="radio://0/68/2M/E7E7E7E702" />
  <arg name="uri3" default="radio://0/35/2M/E7E7E7E703" />
  <arg name="uri4" default="radio://0/80/2M/E7E7E7E704" />
  <arg name="uri5" default="radio://0/57/2M/E7E7E7E705" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <!-- <group ns="crazyflie1">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri1)" />
      <arg name="tf_prefix" value="crazyflie1" />
      <arg name="enable_logging" value="False" />
      <arg name="roll_trim" value="0.1" />
      <arg name="pitch_trim" value="0.1" />
    </include>
    <node name="controller" pkg="crazyflie_demo" type="a_standing_wave_cf1.py" output="screen"> 
      <param name="start_time" command="date +'%d-%m-%Y_%Ih%Mm%S.%N'"/>
    </node>
  </group> -->

  <!-- <group ns="crazyflie2">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri2)" />
      <arg name="tf_prefix" value="crazyflie2" />
      <arg name="enable_logging" value="False" />
      <arg name="roll_trim" value="0.0" />
      <arg name="pitch_trim" value="0.0" />
    </include>
    <node name="controller" pkg="crazyflie_demo" type="a_standing_wave_cf2.py" output="screen"> 
      <param name="start_time" command="date +'%d-%m-%Y_%Ih%Mm%S.%N'"/>
    </node>
  </group> -->

  <group ns="crazyflie3">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri3)" />
      <arg name="tf_prefix" value="crazyflie3" />
      <arg name="enable_logging" value="False" />
      <arg name="roll_trim" value="0.1" />
      <arg name="pitch_trim" value="0.1" />
    </include>
    <node name="controller" pkg="crazyflie_demo" type="a_standing_wave_cf3.py" output="screen"> 
      <param name="start_time" command="date +'%d-%m-%Y_%Ih%Mm%S.%N'"/>
    </node>
  </group>

  <group ns="crazyflie4">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri4)" />
      <arg name="tf_prefix" value="crazyflie4" />
      <arg name="enable_logging" value="False" />
      <arg name="roll_trim" value="-0.6" />
      <arg name="pitch_trim" value="0.1" />
    </include>
    <node name="controller" pkg="crazyflie_demo" type="a_standing_wave_cf4.py" output="screen"> 
      <param name="start_time" command="date +'%d-%m-%Y_%Ih%Mm%S.%N'"/>
    </node>
  </group>

  <group ns="crazyflie5">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri5)" />
      <arg name="tf_prefix" value="crazyflie5" />
      <arg name="enable_logging" value="False" />
      <arg name="roll_trim" value="0.5" />
      <arg name="pitch_trim" value="-0.8" />
    </include>
    <node name="controller" pkg="crazyflie_demo" type="a_standing_wave_cf5.py" output="screen"> 
      <param name="start_time" command="date +'%d-%m-%Y_%Ih%Mm%S.%N'"/>
    </node>
  </group>

  <include file="$(find vicon_bridge)/launch/vicon.launch"/>
</launch>